File: engine/engine-model/src/main/scala/tools/packet/monster/HealMonster.scala
package tools.packet.monster

import net.opcodes.SendOpcode
import tools.packet.PacketInput

class HealMonster(private var _objectId: Int, private var _heal: Int, private var _currentHp: Int, private var _maximumHp: Int) extends PacketInput {
  def objectId: Int = _objectId

  def heal: Int = _heal

  def currentHp: Int = _currentHp

  def maximumHp: Int = _maximumHp

  override def opcode(): SendOpcode = SendOpcode.DAMAGE_MONSTER
}

File: engine/engine-model/src/main/java/tools/packet/monster/HealMonster.java
package tools.packet.monster;

import net.opcodes.SendOpcode;
import tools.packet.PacketInput;

public record HealMonster(Integer objectId, Integer heal, Integer currentHp, Integer maximumHp) implements PacketInput {
   @Override
   public SendOpcode opcode() {
      return SendOpcode.DAMAGE_MONSTER;
   }
}